[identity] log warning for users trying to access cloud shell using user assigned identity (#20226)
Azure/azure-sdk-for-js: sdk/identity/identity/CHANGELOG.md

# Release History

## 2.1.0-beta.1 (Unreleased)

### Features Added

- Added log warning for non-support of user assigned identity in Managed Identity credentials in Cloud Shell environments

### Breaking Changes

### Bugs Fixed

### Other Changes

## 2.0.3 (Unreleased)

### Features Added

- Added log warning for non-support of user assigned identity in Managed Identity credentials in Cloud Shell environments

### Breaking Changes

### Bugs Fixed

> ### Other Changes

Author: ```suggestion
### Other Changes

```

# Release History

> ## 2.1.0-beta.1 (2022-02-08)

Author: ```suggestion
## 2.1.0-beta.1 (Unreleased)
```

### Features Added

> - Added log warning for non-support of user assigned identity in Managed Identity credentials in Cloud Shell environments

Reviewer: ```suggestion
- In the Azure Cloud Shell, if a user assigned identity is provided through the `clientId` parameter to the `ManagedIdentityCredential`, this credential now logs a warning.
```

What do you think of this?

Reviewer: I updated this suggestion.

async function main() {
  const credential = new DeviceCodeCredential({
    tokenCachePersistenceOptions: {
      enabled: true,

>     },

Author: ```suggestion
    }
```

- We have also renamed the error `CredentialUnavailable` to `CredentialUnavailableError`, to align with the naming convention used for error classes in the Azure SDKs in JavaScript.
- In v1 of Identity some `getToken` calls could resolve with `null` in the case the authentication request succeeded with a malformed output. In v2, issues with the `getToken` method will always throw errors.
- Breaking changes to InteractiveBrowserCredential

> 

Reviewer: Is this change necessary?

Author: nope
